analytics/event.go

```diff
@@ -0,0 +1,31 @@
+package analytics
+
+import "time"
+
+// Holds the name and properties of an analytics event.
+type Event struct {
+	// The name of the event.
+	name string
+
+	// The timestamp of the event. Set to nil to use the current time.
+	timestamp time.Time
+
+	// Custom properties of the event.
+	properties map[string]any
+}
+
+// Returns a new event with the given name, properties, and timestamp. Given no timestamp is provided, the current time is used.
+func NewEvent(name string, properties map[string]any, timestamp ...time.Time) *Event {
```

Comment:
I'm not sure what having a separate Event object gives us? Although both Mixpanel and Segment do it, so I guess there must be some good reason.  :)

The expected use case is probably
```
analyticsClient.Track( user.Email, analytics.NewEvent(
    "my event",
    map[string]any{ 
       "cause" : "something",
    },
) )
```

so I wonder if it's better to combine the two in the common case:
```
analyticsClient.Track( user.Email, "my event",
    map[string]any{ 
       "cause" : "something",
    },
)
```

and have a separate method that can take a fully-populated Event.

Reply:
I think it might be helpful for ensuring consistent event types and properties using helper functions. This might be helpful if we were to send the same event in multiple places.

So something like:
```
func NewFooEvent(value string) *Event {
	return NewEvent("foo", map[string]any{
		"foo": value,
	}),
}
```

However, I do agree that it would be nice to have a shorthand for this. I've added a new method to handle both cases in [064e35e](https://github.com/akitasoftware/akita-libs/pull/152/commits/064e35e445b2ee0ac173764b98b99415022e1fca)